test/Validation.test.js

```diff
@@ -255,6 +255,98 @@ describe("Validation", () => {
 				" - configuration.stats should be one of these:"
 			);
 		}
+	}, {
+		name: "Invalid plugin provided: bool",
+		config: {
+			entry: "foo.js",
+			plugins: [
+				false
+			]
+		},
+		message: [
+			" - configuration.plugins[0] should be one of these:",
+			"   object { apply, ... } | function",
+			"   -> Plugin of type object or instanceof Function",
+			"   Details:",
+			"    * configuration.plugins[0] should be an object.",
+			"      -> Compiler or Resolver Plugin instance of Function",
+			"    * configuration.plugins[0] should be an instance of function",
+			"      -> Compiler or Resolver Plugin instance of Function"
+		]
+	}, {
+		name: "Invalid plugin provided: array",
+		config: {
+			entry: "foo.js",
+			plugins: [
+				[]
+			]
+		},
+		message: [
+			" - configuration.plugins[0] should be one of these:",
+			"   object { apply, ... } | function",
+			"   -> Plugin of type object or instanceof Function",
+			"   Details:",
+			"    * configuration.plugins[0] should be an object.",
+			"      -> Compiler or Resolver Plugin instance of Function",
+			"    * configuration.plugins[0] should be an instance of function",
+			"      -> Compiler or Resolver Plugin instance of Function"
+		]
+	}, {
+		name: "Invalid plugin provided: string",
+		config: {
+			entry: "foo.js",
+			plugins: ["abc123"]
+		},
+		message: [
+			" - configuration.plugins[0] should be one of these:",
+			"   object { apply, ... } | function",
+			"   -> Plugin of type object or instanceof Function",
+			"   Details:",
+			"    * configuration.plugins[0] should be an object.",
+			"      -> Compiler or Resolver Plugin instance of Function",
+			"    * configuration.plugins[0] should be an instance of function",
+			"      -> Compiler or Resolver Plugin instance of Function"
+		]
+	}, {
+		name: "Invalid plugin provided: int",
+		config: {
+			entry: "foo.js",
+			plugins: [
+				12
+			]
+		},
+		message: [
+			" - configuration.plugins[0] should be one of these:",
+			"   object { apply, ... } | function",
+			"   -> Plugin of type object or instanceof Function",
+			"   Details:",
+			"    * configuration.plugins[0] should be an object.",
+			"      -> Compiler or Resolver Plugin instance of Function",
+			"    * configuration.plugins[0] should be an instance of function",
+			"      -> Compiler or Resolver Plugin instance of Function"
+		]
+	}, {
+		name: "Invalid plugin provided: object without apply function",
+		config: {
+			entry: "foo.js",
+			plugins: [
+				new function() {}
```

Comment:
This looks weird. Why not just passing `{}`?

Reply:
oh yeah this was used back on functional check as experiment, lemme update that fellow